api/desecapi/serializers.py

```diff
@@ -127,15 +210,16 @@ def validate_type(value):
                 "Generic type format is not supported.")
         return value
 
-    def to_representation(self, instance):
-        data = super().to_representation(instance)
-        data.pop('id')  # TODO remove this by changing Meta subclass
-        return data
+    def existence_condition(self, arg):
+        if isinstance(arg, RRset):
+            return arg.records.exists()
+        else:
+            return bool(arg.get('records')) if 'records' in arg.keys() else True
 
     def create(self, validated_data):
         rrs_data = validated_data.pop('records')
         rrset = RRset.objects.create(**validated_data)
-        rrset.set_all_record_contents(set([rr_data['content'] for rr_data in rrs_data]))
+        rrset.set_all_record_contents(set([rr_data for rr_data in rrs_data]))
```

Comment:
list comprehension is unnecessary, you can just pass `rrs_data`. Also, is the `set()` needed here?